mmv1/products/documentaiwarehouse/DocumentSchema.yaml

```diff
@@ -0,0 +1,265 @@
+              Configurations for an integer property.
+            send_empty_value: true
+        - !ruby/object:Api::Type::NestedObject
+          name: 'floatTypeOptions'
+          description: |
+            Float property.
+          send_empty_value: true
+          allow_empty_object: true
+          properties:
+          - !ruby/object:Api::Type::String
+            name: 'dummy'
+            description: |
+              Configurations for a float property.
+            send_empty_value: true
+        - !ruby/object:Api::Type::NestedObject
+          name: 'textTypeOptions'
+          description: |
+            Text/string property.
+          send_empty_value: true
+          allow_empty_object: true
+          properties:
+          - !ruby/object:Api::Type::String
+            name: 'dummy'
+            description: |
+              Configurations for a text property.
+            send_empty_value: true
+        - !ruby/object:Api::Type::NestedObject
+          name: 'propertyTypeOptions'
+          description: |
+            Nested structured data property.
+          send_empty_value: true
+          properties:
+            - !ruby/object:Api::Type::Array
+              name: 'propertyDefinitions'
+              description: |
+                Defines the metadata for a schema property.
+              item_type: !ruby/object:Api::Type::NestedObject
+                properties:
+                  - !ruby/object:Api::Type::String
+                    name: 'name'
+                    description: |
+                      The name of the metadata property.
+                  - !ruby/object:Api::Type::String
+                    name: 'displayName'
+                    required: true
+                    description: |
+                      The display-name for the property, used for front-end.
+                  - !ruby/object:Api::Type::Boolean
+                    name: 'isRepeatable'
+                    description: |
+                      Whether the property can have multiple values.
+                  - !ruby/object:Api::Type::Boolean
+                    name: 'isFilterable'
+                    description: |
+                      Whether the property can be filtered. If this is a sub-property, all the parent properties must be marked filterable.
+                  - !ruby/object:Api::Type::Boolean
+                    name: 'isSearchable'
+                    description: |
+                      Indicates that the property should be included in a global search.
+                  - !ruby/object:Api::Type::Boolean
+                    name: 'isMetadata'
+                    description: |
+                      Whether the property is user supplied metadata.
+                  - !ruby/object:Api::Type::Boolean
+                    name: 'isRequired'
+                    description: |
+                      Whether the property is mandatory.
+                  - !ruby/object:Api::Type::NestedObject
```

Comment:
Besides `textTypeOptions`, should other property types  (integer, float, text, property, enum, dateTime, map, timestamp) be allowed?

Reply:
Generally no, because other property types make the schema super complex which has not been the case yet for any customer. But it is a possibility in the future. I have added those properties now. Thanks for pointing them out!